Sources/Data Model/Audience/SemanticVersion.swift

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+* Copyright 2020, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+
+import Foundation
+
+typealias SemanticVersion = String
+/*
+ This comparison is ported from the Optimizely web version of semantic version compare.
+ https://github.com/optimizely/client-js/blob/devel/src/core/lib/compare_version.js
```

Comment:
This is a private repo. We should ideally drop the link here.

Reply:
We should probably add a note that beta/alpha/any other string which does not resemble x.y.z will not be evaluated here.

Reply:
@aliabbasrizvi take a closer look at the implementation.  it does allow for string tags.